astropy/io/fits/connect.py

```diff
@@ -286,16 +294,16 @@ def read_table_fits(
             mask = col.array == b""
             fill_value = b""
 
+        arr = data[col.name]
+        if strip_spaces and coltype is np.bytes_:
```

Comment:
Doesn't this come too late? The mask is set above based on array elements being empty, but they could also become empty after stripping. 

EDIT: reading on I think this seems intended: spaces lead to an empty string but not a masked one. Please do add a comment, and a specific test as well!

Reply:
Actually I'm not sure, if we strip spaces after making the mask you can get unmasked empty strings mixed with masked ones, which doesn't make sense... so I would tend to do the stripping before masking?

Reply:
@mhvk - just to check on this one which I think was the main remaining point, should stripping be done before computing the mask ?